typescript/lib/config.ts

```diff
@@ -88,6 +101,8 @@ export class AIConfigRuntime implements AIConfig {
     this.schema_version = schemaVersion;
     this.metadata = metadata || {};
     this.prompts = prompts || [];
+
+    registerDefaultModelParsers();
```

Comment:
I think this could be problematic in two cases:
- User currently registers custom parsers before instantiating an AIConfigRuntime; this could overwrite their custom registry
- Even if they instantiate an AIConfigRuntime *before* customizing the registry, it can still be overwritten the next time they construct an AIConfigRuntime instance

Reply:
Ah, good point. My main issue was the AnyscaleEndpoint model parser takes in an API key in the constructor, which may not have loaded from the env at the start when the registration is done statically.

Reply:
Maybe we can just do a simple check here to see if there are already model parsers registered (registry list is nonempty), and only register these defaults if not?

Reply:
I think that can cause other issues, because if a user has already registered custom model parsers, it won't register the default ones. I've updated this by creating an AnyscaleEndpointModelParser that overrides OpenAIChatModelParser and just updates the openAI client in the `run` function itself.